06-reactJs/components/form.js

```diff
@@ -0,0 +1,48 @@
+var React = require('react');
+
+var Form = React.createClass({
+    getInitialState: function () {
+        return({
+            title: '',
+            genre: '',
+            favorite: false
+        });
+    },
+    render: function () {
+        return (
+            <form onSubmit={this.handleSubmit}>
+                <input type='text' placeholder='Title' refs='title' onChange={this.handleTitleChange} />
+                <input type='text' placeholder='Genre' refs='genre' onChange={this.handleGenreChange} />
+                <input type='checkbox' onClick={this.handleCheckboxChange}>Add to favorites </input>
+                <input type='submit' value='Post' />
+            </form>
+        );
+    },
+
+    handleTitleChange: function (event) {
+        this.setState({title: event.target.value});
```

Comment:
This is not incorrect, but maybe you'll want to update the state only when you submit information instead changing it each time you input data.